src/middle-pgsql.cpp

```diff
@@ -314,6 +314,46 @@ std::size_t middle_query_pgsql_t::get_way_node_locations_db(
     return count;
 }
 
+void middle_pgsql_t::node(osmium::Node const &node)
+{
+    if (node.deleted()) {
+        node_delete(node.id());
+    } else {
+        if (m_options->append) {
+            node_delete(node.id());
+            node_set(node);
+        } else {
+            node_set(node);
```

Comment:
Any particular reason why you don't move the node_set() outside the if? Dto for the other two functions.

Reply:
Just not gotten around to that. I have added a commit now that does that.